monitoring/src/check-page/common-functions.ts

```diff
@@ -46,33 +47,94 @@ export const makeNewBrowser = async (debugMode: boolean): Promise<Browser> => {
+	config: Config,
+	page: Page,
+): Promise<void> => {
+	await page.waitForTimeout(3000);
+	log_info(`Waiting for Privacy Settings Panel: Start`);
+	const frame = getFrame(page, config.iframeDomainSecondLayer);
+	await frame.waitForSelector(ELEMENT_ID.TCFV2_SECOND_LAYER_HEADLINE);
+	log_info(`Waiting for Privacy Settings Panel: Complete`);
+};
+
+export const clickSaveAndCloseSecondLayer = async (
+	config: Config,
+	page: Page,
+) => {
+	log_info(`Clicking on save and exit button: Start`);
+	// Ensure that Sourcepoint has enough time to load the CMP
+	await page.waitForTimeout(5000);
+
+	const frame = getFrame(page, config.iframeDomainSecondLayer);
+	await frame.click(ELEMENT_ID.TCFV2_SECOND_LAYER_SAVE_AND_EXIT);
+
+	log_info(`Clicking on save and exit button: Complete`);
+};
+
+export const clickRejectAllSecondLayer = async (config: Config, page: Page) => {
+	log_info(`Clicking on reject all button: Start`);
+
+	await page.waitForTimeout(5000);
+
+	const frame = getFrame(page, config.iframeDomainSecondLayer);
+
+	await frame.click(ELEMENT_ID.TCFV2_SECOND_LAYER_REJECT_ALL);
+
+	log_info(`Clicking on reject all button: Complete`);
+};
+
+// TODO: consider better approach for getting frame - expensive?
+export const getFrame = (page: Page, iframeUrl: string): Frame => {
+	const frame = page.frames().find((f) => f.url().startsWith(iframeUrl));
+
+	if (frame === undefined) {
+		throw new Error(`Could not find frame ${iframeUrl} : Failed`);
+	}
+
+	return frame;
+};
+
 export const checkTopAdHasLoaded = async (page: Page): Promise<void> => {
 	log_info(`Waiting for ads to load: Start`);
-	await page.waitForSelector(
-		'.ad-slot--top-above-nav .ad-slot__content iframe',
-		{ timeout: 30000 },
-	);
+	await page.waitForSelector(ELEMENT_ID.TOP_ADVERT, { timeout: 30000 });
 	log_info(`Waiting for ads to load: Complete`);
 };
 
 export const checkCMPIsOnPage = async (page: Page): Promise<void> => {
 	log_info(`Waiting for CMP: Start`);
-	await page.waitForSelector('[id*="sp_message_container"]');
-	log_info(`Waiting for CMP: Finish`);
+	await page.waitForSelector(ELEMENT_ID.CMP_CONTAINER);
+	log_info(`Waiting for CMP: Complete`);
 };
 
 export const checkCMPIsNotVisible = async (page: Page): Promise<void> => {
 	log_info(`Checking CMP is Hidden: Start`);
 
-	const getSpMessageDisplayProperty = function () {
-		const element = document.querySelector('[id*="sp_message_container"]');
+	const getSpMessageDisplayProperty = function (selector: string) {
+		const element = document.querySelector(selector);
 		if (element) {
 			const computedStyle = window.getComputedStyle(element);
 			return computedStyle.getPropertyValue('display');
 		}
 	};
-
-	const display = await page.evaluate(getSpMessageDisplayProperty);
+	// When testing AUS locally over a VPN we need to make sure everything has finished
+	// loading before testing for existence of CMP otherwise we are prone to false positives.
+	// await page.waitForTimeout(3000); // for testing AUS over a VPN
```

Comment:
Should this be commented out?

Reply:
Interestingly, I have not needed it recently and PROD CMP Monitoring does not seem to use or need it so I have removed this.